### PR TITLE
docs: fix edge case in rational number conversion `float(a//b)`

### LINKED ISSUE #56726: Documentation about "Conversion from rational to floating-point"
The docs state:

> Conversion from rational to floating-point respects the following identity for any integral values of a and b, with the exception of the two cases b == 0 and a == 0 && b < 0:
> 
> julia> a = 1; b = 2;
> 
> julia> isequal(float(a//b), a/b)
> true

https://docs.julialang.org/en/v1/manual/complex-and-rational-numbers/

I do not understand the first case (`b == 0`), because:
```jl
julia> isequal(float(1//0),1/0)
true
julia> isequal(float(-1//0),-1/0)
true
```

### PR DESCRIPTION
**Pull request by Priynsh**
Fixes #56726 added the changes that were suggested. fixing the mistake.

### CODE REVIEW

File: doc/src/manual/complex-and-rational-numbers.md [hunk 1]
```diff
@@ -254,13 +254,29 @@ julia> float(3//4)
 ```
 
 Conversion from rational to floating-point respects the following identity for any integral values
-of `a` and `b`, with the exception of the two cases `b == 0` and `a == 0 && b < 0`:
+of `a` and `b`, with the exception of the two cases `a==0 && b==0` and `a==0 && b < 0`:
```
Comment (fingolfin) [inline doc/src/manual/complex-and-rational-numbers.md:257]:
```suggestion
of `a` and `b`, with the exception of the case `a==0 && b <= 0`:
```
or maybe
```suggestion
of `a` and `b`, except when `a==0 && b <= 0`:
```

Comment (Priynsh) [reply doc/src/manual/complex-and-rational-numbers.md]: Thanks!

File: doc/src/manual/complex-and-rational-numbers.md [hunk 2]
```diff
@@ -254,13 +254,29 @@ julia> float(3//4)
...
 Conversion from rational to floating-point respects the following identity for any integral values
-of `a` and `b`, with the exception of the two cases `b == 0` and `a == 0 && b < 0`:
+of `a` and `b`, except when `a==0 && b <= 0`:
 
 ```jldoctest
 julia> a = 1; b = 2;
 
 julia> isequal(float(a//b), a/b)
 true
+julia> a, b = 0, 0
```
Comment (KristofferC) [inline doc/src/manual/complex-and-rational-numbers.md:264]: missing empty line?

Comment (Priynsh) [reply doc/src/manual/complex-and-rational-numbers.md:265]: mb, fixed

**Review by inkydragon — APPROVED** [review]

**Pull request merged**